driver-core/src/test/java/com/datastax/driver/core/policies/PerHostErrorTrackerTest.java

```diff
@@ -0,0 +1,256 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core.policies;
+
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.Host;
+import com.datastax.driver.core.HostDistance;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.ScassandraCluster;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.Statement;
+import com.datastax.driver.core.exceptions.UnauthorizedException;
+import org.scassandra.Scassandra;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+import static org.scassandra.http.client.PrimingRequest.queryBuilder;
+import static org.scassandra.http.client.Result.success;
+import static org.scassandra.http.client.Result.unauthorized;
+
+public class PerHostErrorTrackerTest {
```

Comment:
I think that could work.  It would add a little bit of complexity in that we would have to control the calls to `EWMA#tick`, but I agree it could be the right solution.

One problem is that the driver's Clock interface only has `currentTimeMicros()` which is by default `System.currentTimeMillis()*1000`.

The metrics `Clock` has `getTick` which by default associates to `nanoTime` and `getTime` which associates to `currentTimeMillis`.  The main reason you'd want to use `nanoTime` for ticks is that it isn't at the mercy of clock skew (leap seconds, users messing with date, drift, ntp syncs, etc.).

So if we do make this change, I think we should update our `Clock` interface for have 2 separate methods for this intent like the metrics library does.